kale/embed/video_i3d.py

```diff
@@ -375,12 +405,14 @@ def __init__(
             return
 
         end_point = "Logits"
-        self.avg_pool = nn.AvgPool3d(kernel_size=[2, 7, 7], stride=(1, 1, 1))
-        self.avg_pool_flow = nn.AvgPool3d(kernel_size=[1, 7, 7], stride=(1, 1, 1))
+        # self.avg_pool = nn.AvgPool3d(kernel_size=[2, 7, 7], stride=(1, 1, 1))
+        # self.avg_pool_flow = nn.AvgPool3d(kernel_size=[1, 7, 7], stride=(1, 1, 1))
+        self.avg_pool = nn.AdaptiveAvgPool3d(1)
         self.dropout = nn.Dropout(dropout_keep_prob)
         self.logits = Unit3D(
             in_channels=384 + 384 + 128 + 128,
-            output_channels=self._num_classes,
+            output_channels=400,
```

Comment:
Are you sure about these hard codes? Always fixed to such values? This line and above.